src/Panel/Panel.component.tsx

```diff
@@ -1,93 +1,121 @@
-   *  @default '#fff'
-   */
-  borderColor?: string;
-  /** this dictates the border color of the panel
-   *  @default '#eee'
+  panelStyle: string;
+  /** whether the panel can be collapsed
+   *  @default 'false'
    */
-  collapse?: boolean;
-  /** this dictates whether the panel is collapsed or not
+  collapsible?: boolean;
+  /** whether the panel is collapsed or not
    *  @default 'false'
    */
+  isCollapsed?: boolean;
+  /**
+   * From theme provider
+   *
+   * @default defaultTheme
+   **/
+  theme?: any;
 };
 const PanelWrapper = styled.div`
-  border: 1px solid ${(props: Props) => props.borderColor || '#eee'};
-  font-family: '"Open Sans", arial, sans-serif';
+  border: ${(props: Props) =>
+    `${props.theme.borders.width} solid ${props.theme.styles[props.panelStyle]
+      .borderColor || '#eee'}`};
+  overflow: hidden;
+  font-family: ${(props: Props) => props.theme.typography.fontFamily};
+  border-radius: ${(props: Props) => props.theme.borders.radius};
 `;
 const PanelHeader = styled.div`
-  background: ${(props: Props) => props.titleBg};
+  background: ${(props: Props) =>
+    props.theme.styles[props.panelStyle].lightFlood};
   padding: 5px 8px;
-  border-bottom: 1px solid ${(props: Props) => props.borderColor};
+  border-bottom: 1px solid
+    ${(props: Props) => props.theme.styles[props.panelStyle].borderColor};
   font-size: 20px;
-  color: ${(props: Props) => props.titleColor};
+  color: ${(props: Props) => props.theme.styles[props.panelStyle].text};
+  &:hover {
+    background: ${(props: Props) =>
+      props.collapsible
+        ? props.theme.styles[props.panelStyle].hoverLightFlood
+        : 'auto'};
+    cursor: ${(props: Props) => (props.collapsible ? 'pointer' : 'auto')};
+  }
 `;
 const PanelBody = styled.div`
-  background: ${(props: Props) => props.bodyBg};
-  padding: 5px 8px;
+  background: white;
+  overflow: hidden;
+  height: auto;
+  padding: ${(props: Props) => '5px 8px'};
   font-size: 16px;
-  opacity: ${(props: Props) => (props.collapse ? 0 : 1)};
+  opacity: ${(props: Props) => 1};
   transition: all 300ms ease-in-out;
-  transform: ${(props: Props) => (props.collapse ? 'scaleY(0)' : 'scaleY(1)')};
 `;
 
-const initialState = { collapse: false };
+const initialState = {
+  isCollapsed: false,
+};
 type State = Readonly<typeof initialState>;
 
 export class Panel extends React.Component<Props, State> {
+  static Header: React.Component;
+  static Body: React.Component;
+  private bodyRef: React.RefObject<HTMLDivElement>;
+
   constructor(props: Props) {
     super(props);
+    this.bodyRef = React.createRef();
     this.handleToggleCollapse.bind(this);
   }
-  state: State = { ...initialState, collapse: this.props.collapse || false };
+
+  readonly state: State = initialState;
+  localIsCollapsed: boolean = false;
+
+  componentWillReceiveProps(nextProps: Props) {
+    this.localIsCollapsed = !!nextProps.isCollapsed;
+    this.handleToggleCollapse();
```

Comment:
We need to add a check here so that we only run `handleToggleCollapse` if the prop that changed is `isCollapsed`. Otherwise 10px gets added to the body with each prop change.